app/services/search/location_builder.rb

```diff
@@ -71,4 +72,8 @@ def build_location_filter(location, radius)
   def initialize_nationwide_search
     @location = nil
   end
+
+  def map_location_with_incorrect_coordinates
```

Comment:
I find the method name potentially confusing. I know what you mean because I know what the ticket is about. A more explicit method name could be `map_location_name`.

However, given my previous comment, I'd recommend combining the behaviour in line 13 (the first assignment) with this method to make a `initialize_location` method that handles both pieces of assignment logic at once.